src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java

```diff
@@ -97,6 +97,9 @@
   @Autowired
   ApplicationContext myApplicationContext;
 
+  @Autowired(required = false)
+  RepositoryValidationInterceptorFactoryR4 factory;
+
```

Comment:
Does it make sense for the R4 factory to be in BaseJpaREstfulServer? Seems like those repositoryValidationInterceptorFactorys could inherit some interface, and you can autowire the interface.

Reply:
Agree - all the leg work had been done so I've introduced the interface